tests/platform_helper/utils/test_validation.py

```diff
@@ -647,6 +648,47 @@ def test_load_and_validate_config_valid_file(yaml_file):
     assert validated == conf
 
 
+def test_lint_yaml_for_duplicate_keys_fails_when_duplicate_keys_provided(
+    valid_platform_config, fakefs
+):
+    fakefs.create_file(PLATFORM_CONFIG_FILE, contents=yaml.dump(valid_platform_config))
+    # Remove the extensions k:v pair from the platform config - re-added as plain text.
+    valid_platform_config.pop("extensions")
+
+    naughty_extensions_with_duplicate = """
+extensions:
+  connors-favourite-key-but-duplicated:
```

Comment:
I would be inclined to simplify this to just `duplicate-key` for easier reading.

You could actually just define the extension here, then drop it in twice after the `extensions` key when you rebuild the `invalid_platform_config`.